alfred-telemetry-platform/src/main/java/eu/xenit/alfred/telemetry/binder/solr/nodedao/SolrNodeDAOMetrics.java

```diff
@@ -0,0 +1,54 @@
+package eu.xenit.alfred.telemetry.binder.solr.nodedao;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.util.function.ToDoubleFunction;
+
+import org.alfresco.repo.domain.node.NodeDAO;
+import org.alfresco.repo.transaction.RetryingTransactionHelper;
+import org.alfresco.service.transaction.TransactionService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SolrNodeDAOMetrics {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(eu.xenit.alfred.telemetry.binder.solr.nodedao.SolrNodeDAOMetrics.class);
+
+    public static final String SOLR_METRICS_PREFIX = "solr.nodedao";
+    private TransactionService transactionService;
+
+    private NodeDAO nodeDAO;
+    private MeterRegistry registry;
+
+    public SolrNodeDAOMetrics(NodeDAO nodeDAOComponent, TransactionService transactionService,
+                               MeterRegistry registry, boolean enabled) {
+        this.nodeDAO = nodeDAOComponent;
+        this.transactionService = transactionService;
+        this.registry = registry;
+        if (enabled) {
+            registerMetrics();
+        }
+    }
+
+    private void registerMetrics() {
+        LOGGER.info("Registering Solr metrics");
+        registerSolrNodeDAOMetric("maxTxNode", NodeDAO::getMaxNodeId, "number");
+        registerSolrNodeDAOMetric("maxTxnId", NodeDAO::getMaxTxnId, "number");
+        registerSolrNodeDAOMetric("maxTxnCommitTime", NodeDAO::getMaxTxnCommitTime, "number");
```

Comment:
commit-time is not a "number"

Reply:
Using a `TimeGauge` would also be better. It automatically scales to the base unit of time expected by each registry implementation.

random example: https://github.com/micrometer-metrics/micrometer/blob/master/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/cache/GuavaCacheMetrics.java#L102

Reply:
as in: I'm not a fan of that `registerSolrNodeDAOMetric` helper :-)